modules/tracktion_engine/timestretch/tracktion_RubberBand.h

```diff
@@ -0,0 +1,112 @@
+/*
+    ,--.                     ,--.     ,--.  ,--.
+  ,-'  '-.,--.--.,--,--.,---.|  |,-.,-'  '-.`--' ,---. ,--,--,      Copyright 2018
+  '-.  .-'|  .--' ,-.  | .--'|     /'-.  .-',--.| .-. ||      \   Tracktion Software
+    |  |  |  |  \ '-'  \ `--.|  \  \  |  |  |  |' '-' '|  ||  |       Corporation
+    `---' `--'   `--`--'`---'`--'`--' `---' `--' `---' `--''--'    www.tracktion.com
+
+    Tracktion Engine uses a GPL/commercial licence - see LICENCE.md for details.
+*/
+
+#pragma once
+
+#define USE_SPEEX 1
+#define USE_BUILTIN_FFT 1
+#define NO_THREADING 1
+#define NO_TIMING 1
+
+#if __APPLE__
+ #define HAVE_VDSP 1
+#endif
+
+#include "../3rd_party/rubberband/rubberband/RubberBandStretcher.h"
+
+struct Stretcher
+{
+    virtual ~Stretcher() {}
+
+    virtual bool isOk() const = 0;
+    virtual void reset() = 0;
+    virtual bool setSpeedAndPitch(float speedRatio, float semitonesUp) = 0;
+    virtual int getFramesNeeded() const = 0;
+    virtual int getMaxFramesNeeded() const = 0;
+    virtual void processData(const float* const* inChannels, int numSamples, float* const* outChannels) = 0;
+    virtual void flush(float* const* outChannels) = 0;
+};
+
+struct RubberStretcher  : public Stretcher
+{
+    RubberStretcher(double sourceSampleRate, int samplesPerBlock, int numChannels, RubberBand::RubberBandStretcher::Option option)
+                  : thisRubberBandStretcher(sourceSampleRate, numChannels, option, 1.0f, 0.5f)
+
+    {
+
+        sampleRate = sourceSampleRate;
+        TempBuffer.setSize(numChannels, sourceSampleRate);
+
+        if (&thisRubberBandStretcher == nullptr)
```

Comment:
This is a redundant check as a pointer to a member object will never be nullptr. You only need to check _pointers_ if they are `nullptr`.